tests/nvmeof/test_ceph_nvmeof_neg_tests.py

```diff
@@ -193,193 +437,78 @@ def check_client(verify=False):
-        gw_info_bkp = json.loads(gw_info_bkp.strip())["subsystems"]
-
-        # restart nvmeof service
-        group = config.get("gw_group", "")
-        service_name = f"nvmeof.{pool}"
-        if group:
-            service_name = f"{service_name}.{group}"
-        restart_cfg = {
-            "no_cluster_state": False,
-            "config": {
-                "service": service_name,
-                "command": "restart",
-                "args": {"verify": True, "service_name": service_name},
-                "pos_args": [service_name],
-            },
-        }
-        test_orch.run(ceph_cluster, **restart_cfg)
-        gw_info = list_subsystems(**sub_args)
-        gw_info = json.loads(gw_info.strip())["subsystems"]
-
-        if gw_info != gw_info_bkp:
-            raise Exception(
-                f"GW Entities aren't same. Actual: {gw_info_bkp} Current: {gw_info}"
-            )
-        LOG.info("Validation of Ceph-83575467 is successful.")
     except Exception as err:
         raise Exception(err)
-    finally:
-        cleanup_cfg = {
-            "gw_node": config["gw_node"],
-            "disconnect_all": [config["initiator_node"]],
-            "cleanup": ["disconnect_all", "gateway", "pool"],
-            "rbd_pool": pool,
-        }
-        config.update(cleanup_cfg)
-        teardown(ceph_cluster, rbd, nvmegwcli, config)
 
 
-def test_ceph_83576085(ceph_cluster, rbd, pool, config):
+def test_ceph_83576085(ceph_cluster, rbd, nvme_service, pool, config):
     """CEPH-83576085: Perform map and unmap NVMe namespaces in loop."""
-    gw_node = get_node_by_id(ceph_cluster, config["gw_node"])
-    ceph = Orch(ceph_cluster, **{})
-    nvmegwcli = create_gateway(
-        nvme_gw_cli_version_adapter(ceph_cluster),
-        gw_node,
-        mtls=config.get("mtls"),
-        shell=getattr(ceph, "shell"),
-        port=config.get("gw_port", 5500),
-        gw_group=config.get("gw_group"),
-    )
-
-    subsystem = dict()
-    listener_port = find_free_port(gw_node)
-    subsystem.update(
-        {
-            "nqn": "nqn.2016-06.io.spdk:ceph-83576085",
-            "serial": 113,
-            "listener_port": listener_port,
-            "allow_host": "*",
+    client = get_node_by_id(ceph_cluster, config["initiator_node"])
+    initiator = NVMeInitiator(client)
+    try:
+        # Deploy NVMe Service
+        if config.get("install"):
+            nvme_service.deploy()
+
+        # Initialize gateways
+        nvme_service.init_gateways()
+        nvmegwcli = nvme_service.gateways[0]
+        cmd_args = {
+            "transport": "tcp",
+            "traddr": nvmegwcli.node.ip_address,
+            "trsvcid": config["initiators"][0]["listener_port"],
         }
-    )
 
-    initiator_cfg = {
-        "subnqn": "nqn.2016-06.io.spdk:ceph-83576085",
-        "listener_port": listener_port,
-        "node": config["initiator_node"],
-    }
+        # Configure Subsystems
+        configure_subsystems(nvme_service)
```

Comment:
Can we not directly call configure_gw_entities and that will take care of configuring subsystems, listeners and hosts right? All you need to do is pass those keywords to config I guess. Can we check this?